Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -9,8 +9,14 @@
 #include <AK/CharacterTypes.h>
 #include <AK/Debug.h>
 #include <AK/HashMap.h>
+#include <AK/Utf8View.h>
+#include <LibUnicode/CharacterTypes.h>
 #include <stdio.h>
 
+#ifdef ENABLE_UNICODE_DATA
+#   include <LibUnicode/UnicodeData.h>
+#endif
```

Comment:
Copied this pattern from some other place. Is this how we are supposed to guard any unicode data usage? (cc @trflynn89 perhaps)

Reply:
Oh, hmm, so I was trying to keep the `#ifdef` mess completely hidden inside LibUnicode's .cpp files, but I see that doesn't quite work when callers want to refer to the generated enums by their values directly...

I'll think about how to make this nicer for user cases like this. In the meantime, to avoid the `#ifdef` spaghetti, you could do something like:
```c++
bool Lexer::is_whitespace() const
{
    ...
    
    static auto space_separator = Unicode::general_category_from_string("Space_Separator"sv);
    if (space_separator.has_value())
        return Unicode::code_point_has_general_category(codepoint, *space_separator);
    return false;
}
```

Reply:
Yeah seems hard to avoid if you want to use the enum directly.
I'll change it to that pattern for now then. Thanks!

Reply:
Oh, semi unrelated, but IIRC at some point someone went through the code base and standardized "code point" being two words. So `codepoint` var names should be `code_point` to match that :)

Reply:
Hmm seems some of the categories are not available for both `ID_Start` and `ID_Continue`. Am I doing something wrong?

Reply:
I had the same direct access issue when i was briefly trying to update libgfx's unicode bidi implementation to use LibUnicode, so +1 for that

Reply:
`ID_Start` and `ID_Continue` are properties, so use `Unicode::property_from_string` for those (if you aren't already)

Reply:
> Oh, semi unrelated, but IIRC at some point someone went through the code base and standardized "code point" being two words. So `codepoint` var names should be `code_point` to match that :)

Even my commit was inconsistent so was already fixing it. Thanks for the good eye